src/WebJobs.Script/Config/RpcWorkerConcurrencyOptions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class RpcWorkerConcurrencyOptions : Hosting.IOptionsFormatter
+    {
+        public RpcWorkerConcurrencyOptions()
+        {
+            // Setting deafault values
+            LatencyThreshold = TimeSpan.FromSeconds(1);
+            AdjustmentPeriod = TimeSpan.FromSeconds(10);
+            CheckInterval = TimeSpan.FromSeconds(1);
+            HistorySize = 10;
+            HistoryThreshold = 1F;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether language worker concurrancy
+        /// is enabled.
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the latency threshold dictating when language worker channel is overloaded.
+        /// </summary>
+        public TimeSpan LatencyThreshold { get; set; }
+
+        /// <summary>
+        /// Gets or sets the time when language worker
+        /// channels will start to monitor again after adding a new worker.
+        /// </summary>
+        public TimeSpan AdjustmentPeriod { get; set; }
+
+        /// <summary>
+        /// Gets or sets interval to check language worker channel state.
+        /// </summary>
+        public TimeSpan CheckInterval { get; set; }
+
+        /// <summary>
+        /// Gets or sets the history size to store language workers channel states.
+        /// </summary>
+        public int HistorySize { get; set; }
```

Comment:
I don't think we should expose some of these publicly - probably more than we want users to have to think about. E.g. AdjustmentPeriod, CheckInterval, HistorySize, HistoryThreshold could all be internal for now I think. It's our job to determine the right numbers for these that work.

Reply:
These settings are exposed publicly for experimenting without redeploying, once we find the best values they will become internal.

Reply:
I see - so you're saying that you won't check them in as public and this is just draft for your testing?

Reply:
We want to test in production after the PR merge and have ability to change the settings. After we the testing is done we can hardcode the values with fixed values.